pkg/registry/registry_quay.go

```diff
@@ -0,0 +1,126 @@
+package registry
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	nurl "net/url"
+	"regexp"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/retagger/pkg/images"
+)
+
+// This code is largely lifted from nokia/docker-registry-client. It provides Quay-specific API behavior needed by retagger.
```

Comment:
I understand this, but we should refactor it to fit better.

Reply:
Agree entirely